File: java/po/world_simulator/entities/animals/Antelope.java
package po.world_simulator.entities.animals;

import po.world_simulator.entities.Entity;

import java.util.Map;

import po.world_simulator.Vector2;
import po.world_simulator.World;
import po.world_simulator.entities.Animal;

public class Antelope extends Animal {
    public Antelope(Vector2 position, int power, int age) {
        super(power, 4, position, "Antelope", age, false);
    }

    public Antelope(Vector2 position) {
        this(position, 4, 0);
    }

    public Antelope(Map<String, String> entityData){
        this(
            new Vector2(entityData.get("position")),
            Integer.parseInt(entityData.get("power")),
            Integer.parseInt(entityData.get("age"))
        );
    }

    @Override
    protected void move(Vector2 moveDirection) {
        // 2 tile move
        moveDirection.multiplySelf(2);

        if (!World.getMap().isTileOutOfBounds(this.position.add(moveDirection))) {
            super.move(moveDirection);
        }
    }

    @Override
    public void collision(Entity otherEntity) {
        if (!this.getClass().equals(otherEntity.getClass())) {
            if (Math.random() < 0.5) {
                if (this.runAway()) {
                    return;
                }
            }
        }
        super.collision(otherEntity);
    }
}

File: python/po/world_simulator/entities/animals/Antelope.py
import random
from po.world_simulator.entities.Animal import Animal
from po.world_simulator.Vector2 import Vector2
import po.world_simulator.World as world_module

class Antelope(Animal):
    # [AI] Merged constructors using parameter types.
    def __init__(self, position_or_data, power=4, age=0):
        if isinstance(position_or_data, dict):
            super().__init__(
                int(position_or_data.get("power", 0)),
                4,
                Vector2(position_or_data.get("position")),
                "Antelope",
                int(position_or_data.get("age", 0)),
                False
            )
        else:
            super().__init__(power, 4, position_or_data, "Antelope", age, False)

    def move(self, moveDirection):
        # 2 tile move
        moveDirection.multiplySelf(2)

        if not world_module.World.getMap().isTileOutOfBounds(self.position.add(moveDirection)):
            super().move(moveDirection)

    def collision(self, otherEntity):
        if type(self) != type(otherEntity):
            if random.random() < 0.5:
                if self.runAway():
                    return
        super().collision(otherEntity)
